packages/metal-component/src/Component.js

```diff
@@ -27,13 +27,32 @@ import { EventEmitter, EventHandler } from 'metal-events';
  * class CustomComponent extends Component {
  *   created() {
  *   }
- *
+
  *   rendered() {
  *   }
  *
+ *   willAttach() {
+ *   }
+ *
  *   attached() {
  *   }
  *
+ *   willReceiveState() {
+ *   }
+ *
+ *   // JSX components only
```

Comment:
This comment here is a bit confusing... for a while I assumed everything below this point was related to JSX components only and not just this method... maybe place the comment inside? or make it more explicit, like `willReceiveProps is only available in JSX components`?